jda/src/main/java/co/aikar/commands/JDACommandContexts.java

```diff
@@ -66,7 +66,11 @@ public JDACommandContexts(JDACommandManager manager) {
             if (c.hasAnnotation(SelfUser.class)) {
                 return jda.getSelfUser();
             }
-            String arg = c.popFirstArg(); // we pop because we are only issuer aware if we are annotated
+            String arg = c.getFirstArg();
+            if (c.hasAnnotation(Optional.class) && (arg == null || arg.isEmpty())) {
```

Comment:
This should be c.isOptional()

Reply:
Oh is the annotation processed in core?

Reply:
In Bukkit, it's checking against whether the context has the annotation.

Reply:
Feel free to update the other refs for me then :)

The goal is, that only the init process cares about actual annotations, which is ran through processors. 

then runtime just checks parsed results from the initialization.

it avoids reflection api's during runtime.

also, the annotation processor may make a custom annotation define something as Optional w/o it using the Optional annotation. so for accuracy, it NEEDS c.isOptional()

Reply:
Sounds good, I'll edit soon